File: OOPJ/Assignments/Assignment-03/MergeSortedArrays.java
import java.util.Arrays;
class MergeSortedArrays {
    public static void merge(int[] A, int[] B) {
        int m = A.length, n = B.length;
        int i = m - 1, j = 0;

        while (i >= 0 && j < n) {
            if (A[i] > B[j]) {
                int temp = A[i];
                A[i] = B[j];
                B[j] = temp;
            }
            i--;
            j++;
        }

        Arrays.sort(A);
        Arrays.sort(B);
    }

    public static void main(String[] args) {
        int[] A = {1, 3, 5, 7};
        int[] B = {2, 4, 6, 8};

        merge(A, B);
        System.out.println("Merged A: " + Arrays.toString(A));
        System.out.println("Merged B: " + Arrays.toString(B));
    }
}

File: OOPJ/Assignments/Assignment-03/MergeSortedArrays.js
import java.util.Arrays;
class MergeSortedArrays {
    public static void mergeSortedArrays(int[] A, int[] B) {
        int p = A.length, q = B.length;
        int[] merged = new int[p + q];

        System.arraycopy(A, 0, merged, 0, p);
        System.arraycopy(B, 0, merged, p, q);

        Arrays.sort(merged);

        System.arraycopy(merged, 0, A, 0, p);
        System.arraycopy(merged, p, B, 0, q);

        System.out.println("A: " + Arrays.toString(A));
        System.out.println("B: " + Arrays.toString(B));
    }

    public static void main(String[] args) {
        int[] A = {1, 5, 6, 7, 8, 10};
        int[] B = {2, 4, 9};

        mergeSortedArrays(A, B);
    }
}
